src/Client/HttpExceptionHandler.php

```diff
@@ -73,7 +73,10 @@ public function transformResponseToException(RequestInterface $request, Response
      */
     protected function getResponseMessage(ResponseInterface $response)
     {
-        $decodedBody = json_decode($response->getBody()->getContents(), true);
+        $responseBody = $response->getBody();
+        $decodedBody = json_decode($responseBody->getContents(), true);
+
+        $responseBody->rewind();
```

Comment:
You should do it before reading the message, not after. Prevent the bug before reading, not after.